plugins/pluginmanagers/lz-n.nix

```diff
@@ -0,0 +1,106 @@
+        };
+        enabledInSpec = helpers.mkNullOrStrLuaFnOr bool ''
+          When false, or if the function returns false, then this plugin will not be included in the spec.
+          This option corresponds to the `enabled` property of lz.n.
+        '';
+        beforeAll = helpers.mkNullOrLuaFn "Always executed before any plugins are loaded.";
+        before = helpers.mkNullOrLuaFn "Executed before this plugin is loaded.";
+        after = helpers.mkNullOrLuaFn "Executed after this plugin is loaded.";
+        event = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+        cmd = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on command.";
+        ft = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on filetype.";
+        colorscheme = helpers.mkNullOrOption (either str (listOf str)) "Lazy-load on colorscheme.";
+        keys = helpers.mkNullOrOption (listOf helpers.keymaps.mapOptionSubmodule) "Lazy-load on key mapping. Use the same format as `config.keymaps`.";
+        priority = helpers.mkNullOrOption number ''
+          Only useful for start plugins (not lazy-loaded) to force loading certain plugins first. 
+          Default priority is 50 (or 1000 if colorscheme is set).
+        '';
+        load = helpers.mkNullOrLuaFn "Can be used to override the `vim.g.lz_n.load()` function for this plugin.";
+      };
+    };
+in
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lz-n";
+  originalName = "lz.n";
+  maintainers = [ helpers.maintainers.psfloyd ];
+  defaultPackage = pkgs.vimPlugins.lz-n;
+
+  settingsDescription = ''
+    Options provided to `vim.g.lz_n`.
+
+    `{ load = "fun"; }` -> `vim.g.lz_n = { load = fun, }`
+  '';
+
+  settingsOptions = {
+    load = helpers.defaultNullOpts.mkLuaFn "vim.cmd.packadd" ''
+      Function used by `lz.n` to load plugins.
+    '';
+  };
+
+  callSetup = false; # Does not use setup
+
+  extraOptions = {
+    plugins = mkOption {
+      description = "List of plugins processed by lz.n";
+      default = [ ];
+      type = types.listOf lzPluginType;
+    };
+  };
+
+  extraConfig =
+    cfg:
+    let
+      processKeymaps =
+        keymaps:
+        if keymaps == null then
+          null
+        else
+          map (
+            keymap:
+            {
+              __unkeyed_1 = keymap.key;
+              __unkeyed_2 = keymap.action;
+              inherit (keymap) mode;
+            }
+            // keymap.options
+          ) keymaps;
+      processPlugin =
+        plugin:
+        removeAttrs plugin [
+          "name"
+          "enabledInSpec"
+        ]
+        // {
+          __unkeyed = plugin.name;
+          enabled = plugin.enabledInSpec;
+          keys = processKeymaps plugin.keys;
+        };
+      plugins = map processPlugin cfg.plugins;
+      pluginSpecs = if length plugins == 1 then head plugins else plugins;
```

Comment:
We can avoid _this_ specific bit of magic here by moving it to the option's `apply` function:

```nix
apply = v: if length plugins == 1 then head plugins else plugins;
```

Although, IDK if that's a good idea... It would affect what users see when they lookup `config.plugins.lz-n.plugins`.

Reply:
If we're using a coercedTo, then we're already modifying the user's definition. But it's probably better if we can keep things simple and document the option as always being a list (in the final merged config)?

Ideally, we'd get upstream to accept singleton list tables, however for now I think this one processing step is ok.